### PR TITLE
Documenting new REST apis

### PR DESCRIPTION
**Pull request by OGAWAHirofumi**
done by current my sphinx knowledge. BTW, current theme(?) looks like too unreadable. Especially

```
title
=====
```

and 

```
sub title
---------
```

are very similar rendering.

### CODE REVIEW

File: docs/source/rest.rst [hunk 1]
```diff
@@ -5,16 +5,38 @@ Using bepasty's REST-API
 The Rest-API enables you to upload and download files, as well as retrieve informations
 about the file on the server.
 
-Currently (Version 0.3) the REST API provides four API Endpoints::
+Currently (Version |release|) the REST API provides the following API Endpoints::
```
Comment (ThomasWaldmann) [inline docs/source/rest.rst:8]: The docs are versioned with releases, so the version reference here could be also removed completely.

File: docs/source/rest.rst [hunk 2]
```diff
@@ -198,3 +226,92 @@ Example Response::
     Content-Range: bytes 0-150225567/150225567
 
 Opens up a stream and delivers the binary data directly. The above headers can be found in the HTTP Response.
+
+
+Modifying a file information
```
Comment (ThomasWaldmann) [inline docs/source/rest.rst:231]: either:

Modifying file information

or

Modifying a file's information

or even better

... metadata

File: docs/source/rest.rst [hunk 3]
```diff
@@ -198,3 +226,92 @@ Example Response::
...
+============================
+API Interface::
+
+    POST /apis/rest/items/<itemname>/modify
+
+Modify a file information specified by ``<itemname>``. (required permission: MODIFY)
```
Comment (ThomasWaldmann) [inline docs/source/rest.rst:237]: same here

File: docs/source/rest.rst [hunk 4]
```diff
@@ -198,3 +226,92 @@ Example Response::
...
+**itemname**
+    The itemname of the target file.
+
+**Content-Type**
+    The content-type header must be ``application/json``
+
+New metadata is specified by JSON in request body.  Currently this API
+is supporting to modify ``filename`` and ``type``.  For example, if
+you want to modify filename::
+
```
Comment (ThomasWaldmann) [inline docs/source/rest.rst:251]: in the request body

to modify the filename

File: docs/source/rest.rst [hunk 5]
```diff
@@ -198,3 +226,92 @@ Example Response::
...
+    {"filename": "new-filename.txt"}
+
+when you want to modify both of filename and type::
```
Comment (ThomasWaldmann) [inline docs/source/rest.rst:254]: If ... both filename and type::

File: docs/source/rest.rst [hunk 6]
```diff
@@ -198,3 +226,92 @@ Example Response::
...
+
+    {"filename": "new-filename.txt", "type": "new-mimetype"}
+
+POST Response by the server
+---------------------------
+If success, status code == 200. Otherwise status code != 200
```
Comment (ThomasWaldmann) [inline docs/source/rest.rst:260]: On success ...

add a dot at the end.

File: docs/source/rest.rst [hunk 7]
```diff
@@ -198,3 +226,92 @@ Example Response::
...
+
+POST Request by the client
+--------------------------
+
+**itemname**
+    The itemname of the target file.
+
+POST Response by the server
+---------------------------
+If success, status code == 200. Otherwise status code != 200
```
Comment (ThomasWaldmann) [inline docs/source/rest.rst:279]: same

File: docs/source/rest.rst [hunk 8]
```diff
@@ -198,3 +226,92 @@ Example Response::
...
```
Comment (ThomasWaldmann) [inline docs/source/rest.rst:298]: same

File: docs/source/rest.rst [hunk 9]
```diff
@@ -198,3 +226,92 @@ Example Response::
...
```
Comment (ThomasWaldmann) [inline docs/source/rest.rst:317]: same

**Review by ThomasWaldmann — COMMENTED** [review]
thanks for adding docs.

some suggestions / grammar improvements.

**Comment by OGAWAHirofumi** [comment]
hm, this ping pong waste time for both side. it would be faster if you tweaked by yourself...

**Comment by OGAWAHirofumi** [comment]
changes

- fix pointed docs as usual :)
- add link to permissions
- reformat to make more readable

you may not need 3rd one, then let me know or you can cherry-pick.

thanks.

**Comment by ThomasWaldmann** [comment]
well, i hope that is not a total waste of time as we both might learn something / do it better than one alone.

**Comment by ThomasWaldmann** [comment]
like, for example, never mixing a reformatting change into a functional (code) or content (docs) change.

it just makes stuff hard to review and can be easily done separately.

**Comment by ThomasWaldmann** [comment]
about the theme issues you mentioned: we can change this after next release, if we find something prettier.

iirc, we have a different theme for borgbackup, maybe that one?

https://borgbackup.readthedocs.io/en/stable/

**Comment by OGAWAHirofumi** [comment]
> about the theme issues you mentioned: we can change this after next release, if we find something prettier.
> 
> iirc, we have a different theme for borgbackup, maybe that one?
> 
> https://borgbackup.readthedocs.io/en/stable/

I tried some themes, and noticed it is fundamental policy of sphinx to use it as section. and all is not readable. it is why 3rd patch was introduced.

**Comment by ThomasWaldmann** [comment]
https://github.com/bepasty/bepasty-server/pull/251/commits/603ac2645a69e3224091d119c0303750ee0d7b92

such changes are just bad to review. guess you didn't change as much as it looks like, but github diff display is just bad for this.

of course multiple PRs take a bit more human "iowait" time, but make it easier to look at the overall diff of the PR, so it is less human "cpu" time.

**Comment by ThomasWaldmann** [comment]
merged as is, can be improved later.

**Comment by OGAWAHirofumi** [comment]
> such changes are just bad to review. guess you didn't change as much as it looks like, but github diff display is just bad for this.

eh, you are not checked it by shpinx? it didn't mean reformat of rst, it was reformat for html rednering. so outputed html will be changed.
 
> of course multiple PRs take a bit more human "iowait" time, but make it easier to look at the overall diff of the PR, so it is less human "cpu" time.

it is only view from your side. my side needs additional works.

**Comment by ThomasWaldmann** [comment]
for review purposes, i need a diff.

i didn't review rendered docs, did you check if html output looks ok?

**Comment by OGAWAHirofumi** [comment]
> for review purposes, i need a diff.
> 
> i didn't review rendered docs, did you check if html output looks ok?

of course, i checked. but i dont know if you like it, i like though.

**Pull request merged**